lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -721,7 +721,7 @@ void HdVP2Material::Sync(
             // Skip creating a new shader instance if the token is unchanged. There is no plan to
             // implement fine-grain dirty bit in Hydra for the same purpose:
             // https://groups.google.com/g/usd-interest/c/xytT2azlJec/m/22Tnw4yXAAAJ
-            if (_surfaceNetworkToken != token) {
+            if (_surfaceNetworkToken != token && !vp2BxdfNet.nodes.empty()) {
```

Comment:
I think it is better to put the protection test before _ApplyVP2Fixes. The new if statement should protect lines from _ApplyVP2Fixes to the end of the higher-level if block.